src/Symfony/Component/Utf8/Bytes.php

```diff
@@ -0,0 +1,505 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Utf8;
+
+use Symfony\Component\Utf8\Exception\InvalidArgumentException;
+
+final class Bytes implements GenericStringInterface, \Countable
+{
+    private $string = '';
+
+    public static function fromString(string $string): self
+    {
+        $bytes = new self();
+        $bytes->string = $string;
+
+        return $bytes;
+    }
+
+    public static function fromCharCode(int ...$codes): self
+    {
+        $bytes = new self();
+        $bytes->string = implode('', array_map('chr', $codes));
+
+        return $bytes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __toString()
+    {
+        return $this->string;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count(): int
+    {
+        return strlen($this->string);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function length(): int
+    {
+        return strlen($this->string);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEmpty(): bool
+    {
+        return '' === $this->string;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function explode(string $delimiter, int $limit = null): array
+    {
+        if ('' === $delimiter) {
+            throw new InvalidArgumentException('Passing an empty delimiter is not supported by this method. Use getIterator() method instead.');
+        }
+
+        $chunks = array();
+        foreach (explode($delimiter, $this->string, $limit ?: PHP_INT_MAX) as $i => $string) {
+            $chunks[$i] = $chunk = clone $this;
+            $chunk->string = $string;
+        }
+
+        return $chunks;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toLowerCase(): self
+    {
+        $current = setlocale(LC_CTYPE, '0');
+        setlocale(LC_CTYPE, 'C');
+
+        $result = clone $this;
+        $result->string = strtolower($this->string);
+
+        setlocale(LC_CTYPE, $current);
+
+        return $result;
```

Comment:
AFAIK, only the YAML and ExpressionLanguage components have some code dealing with setlocale, because they implement parsers. And adding a common dependencies to these components just for this simple code is not worth it IMO.